dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs

```diff
@@ -10,7 +11,8 @@ namespace Microsoft.SemanticKernel;
 /// Represents a <see cref="FunctionChoiceBehavior"/> that provides either all of the <see cref="Kernel"/>'s plugins' functions to AI model to call or specified ones.
 /// This behavior allows the model to decide whether to call the functions and, if so, which ones to call.
 /// </summary>
-internal sealed class AutoFunctionChoiceBehavior : FunctionChoiceBehavior
+[Experimental("SKEXP0001")]
```

Comment:
Consider making the constructors internal, so that even though the class is public, we don't have to worry about keeping the constructors backward compatible.

Reply:
I tried that and ended up with reflection-based code to help the YamlDotNet serializer instantiate those classes, as it doesn't support non-public constructors out of the box. Then I thought we might eventually end up making those constructors public anyway, because for Yaml deserialization in Native-AOT, we might need to use a source code generation equivalent. Let me try to make the change, and let's discuss it tomorrow.